src/Common/ClipboardButton/ClipboardButton.tsx

```diff
@@ -25,73 +25,87 @@ import { ReactComponent as Check } from '../../Assets/Icon/ic-check.svg'
  * @param iconSize - size of svg icon to be shown, default 16 (icon-dim-16)
  */
 export default function ClipboardButton({
     content,
     copiedTippyText = 'Copied!',
     duration = 1000,
-    trigger,
-    setTrigger = noop,
+    copyToClipboardPromise,
     rootClassName = '',
     iconSize = 16,
 }: ClipboardProps) {
     const [copied, setCopied] = useState<boolean>(false)
-    const [enableTippy, setEnableTippy] = useState<boolean>(false)
+    const setCopiedFalseTimeoutRef = useRef<ReturnType<typeof setTimeout>>(-1)
 
-    const handleTextCopied = () => {
+    const handleTriggerCopy = () => {
         setCopied(true)
+
+        setCopiedFalseTimeoutRef.current = setTimeout(() => {
+            setCopied(false)
+
+            setCopiedFalseTimeoutRef.current = -1
+        }, duration)
     }
-    const isTriggerUndefined = typeof trigger === 'undefined'
 
-    const handleEnableTippy = () => setEnableTippy(true)
-    const handleDisableTippy = () => setEnableTippy(false)
-    const handleCopyContent = useCallback(
-        (e?) => {
-            if (e) stopPropagation(e)
-            copyToClipboard(content, handleTextCopied)
-        },
-        [content],
-    )
-    const iconClassName = `icon-dim-${iconSize} dc__no-shrink`
+    const handleAwaitCopyToClipboardPromise = async (shouldRunCopy?: boolean) => {
+        try {
+            if (shouldRunCopy) {
+                await copyToClipboard(content)
+            } else {
+                await copyToClipboardPromise
+            }
 
-    useEffect(() => {
-        if (!copied) return
+            handleTriggerCopy()
+        } catch {
+            noop()
+        }
+    }
 
-        const timeout = setTimeout(() => {
-            setCopied(false)
-            setTrigger(false)
-        }, duration)
+    const handleCopyContent = async (e?: React.MouseEvent) => {
+        if (e) {
+            stopPropagation(e)
+        }
+
+        await handleAwaitCopyToClipboardPromise(true)
+    }
+
+    useEffect(() => {
+        if (!copyToClipboardPromise) {
+            return
+        }
 
-        return () => clearTimeout(timeout)
-    }, [copied, duration, setTrigger])
+        handleAwaitCopyToClipboardPromise().catch(noop)
+    }, [copyToClipboardPromise])
 
     useEffect(() => {
-        if (!isTriggerUndefined && trigger) {
-            setCopied(true)
-            handleCopyContent()
+        return () => {
+            if (setCopiedFalseTimeoutRef.current > -1) {
+                clearTimeout(setCopiedFalseTimeoutRef.current)
+            }
         }
-    }, [trigger, handleCopyContent])
+    }, [])
+
+    const iconClassName = `icon-dim-${iconSize} dc__no-shrink`
+
     return (
-        <Tippy
-            className="default-tt"
-            content={copied ? copiedTippyText : 'Copy'}
-            placement="bottom"
-            visible={copied || enableTippy}
-            arrow={false}
+        <Tooltip
+            content={'Copy'}
```

Comment:
Should be content="Copy" and why are we nesting tooltips? can we move this tooltip inside button or vice-versa?